runtime/gc_glue_java/ConcurrentMarkingDelegate.cpp

```diff
@@ -450,4 +450,31 @@ MM_ConcurrentMarkingDelegate::concurrentClassMark(MM_EnvironmentBase *env, bool
 }
 #endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
 
+void *
+MM_ConcurrentMarkingDelegate::sealAllocCache(MM_EnvironmentBase *env, MM_TLHAllocationSupport *tlh)
+{
+	J9JavaVM *javaVM = (J9JavaVM *)env->getLanguageVM();
+
+	uint8_t *lastCacheObj = (uint8_t *) tlh->getAlloc();
+	tlh->setAlloc((void *)((uintptr_t) lastCacheObj + OMR_MINIMUM_OBJECT_SIZE));
+
+	J9Class *clazz = J9VMJAVALANGOBJECT_OR_NULL(javaVM);
+	uintptr_t const headerSize = J9JAVAVM_OBJECT_HEADER_SIZE(javaVM);
+	j9object_t instance = (j9object_t) lastCacheObj;
+
+	memset(instance, 0, headerSize);
+
+	if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)) {
+		J9ObjectCompressed *objectHeader = (J9ObjectCompressed*) instance;
+		objectHeader->clazz = (uint32_t)(uintptr_t)clazz;
+	} else {
+		J9ObjectFull *objectHeader = (J9ObjectFull*) instance;
+		objectHeader->clazz = (uintptr_t)clazz;
+	}
+
+	Assert_MM_true(J9GC_J9OBJECT_CLAZZ(lastCacheObj, env) == J9VMJAVALANGOBJECT_OR_NULL(javaVM));
```

Comment:
Is it necessary to call `J9VMJAVALANGOBJECT_OR_NULL()` again? Can `clazz` to be used instead?

Reply:
Yup, should be using clazz instead